pysite/database.py

```diff
@@ -20,24 +23,456 @@ def __init__(self, loop_type: str = "gevent"):
+
+        :param table_name: The name of the table to drop
+        :return: True if the table was dropped, False if the table doesn't exist
+        """
+
+        with self.get_connection() as conn:
+            all_tables = rethinkdb.db(self.database).table_list().run(conn)
+
+            if table_name not in all_tables:
+                return False
+
+            rethinkdb.db(self.database).table_drop(table_name).run(conn)
+            return True
+
+    def query(self, table_name: str) -> Table:
+        """
+        Get a RethinkDB table object that you can run queries against
+
+        >>> db = RethinkDB()
+        >>> query = db.query("my_table")
+        >>> db.run(query.insert({"key": "value"}), coerce=dict)
+        {
+            "deleted": 0,
+            "errors": 0,
+            "inserted": 1,
+            "replaced": 0,
+            "skipped": 0,
+            "unchanged": 0
+        }
+
+        :param table_name: Name of the table to query against
+        :return: The RethinkDB table object for the table
+        """
+
+        return rethinkdb.table(table_name)
+
+    def run(self, query: RqlMethodQuery, *, new_connection: bool=False,
+            connect_database: bool=True, coerce: type=None) -> Union[rethinkdb.Cursor, List, Dict, object]:
+        """
+        Run a query using a table object obtained from a call to `query()`
+
+        >>> db = RethinkDB()
+        >>> query = db.query("my_table")
+        >>> db.run(query.insert({"key": "value"}), coerce=dict)
+        {
+            "deleted": 0,
+            "errors": 0,
+            "inserted": 1,
+            "replaced": 0,
+            "skipped": 0,
+            "unchanged": 0
+        }
+
+        Note that result coercion is very basic, and doesn't really do any magic. If you want to be able to work
+        directly with the result of your query, then don't specify the `coerce` argument - the object that you'd
+        usually get from the RethinkDB API will be returned instead.
+
+        :param query: The full query to run
+        :param new_connection: Whether to create a new connection or use the current request-bound one
+        :param connect_database: If creating a new connection, whether to connect to the database immediately
+        :param coerce: Optionally, an object type to attempt to coerce the result to
+
+        :return: THe result of the operation
+        """
+
+        if not new_connection:
+            result = query.run(self.conn)
+        else:
+            result = query.run(self.get_connection(connect_database))
+
+        if coerce:
+            return coerce(result) if result else coerce()
+        return result
+
+    # endregion
+
+    # region: RethinkDB wrapper functions
+
+    def insert(self, table_name: str, *objects: Dict[str, Any],
+               durability: str="hard", return_changes: Union[bool, str]=False,
+               conflict: Union[str, Callable[[Dict[str, Any], Dict[str, Any]], Dict[str, Any]]]="error"
```

Comment:
This line is reaaaally dense with all these annotations. is there any way to make it less terrible to read?